hal/watchdog_api.h

```diff
@@ -0,0 +1,171 @@
+/** \addtogroup hal */
+/** @{*/
+/* mbed Microcontroller Library
+ * Copyright (c) 2017 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef MBED_WATCHDOG_API_H
+#define MBED_WATCHDOG_API_H
+
+#if DEVICE_WATCHDOG
+
+#if !(DEVICE_RESET_REASON)
+  #error "Watchdog feature depends on reset reason API also being implemented"
```

Comment:
why does a hal api have a dependency? Surely this should be part of the watchdog hal api

Reply:
There's no real dependence, other than the function `watchdog_caused_last_reset` which I think should be removed as that can be handled in the platform API